toc/rfc/rfc-draft-ipv6-dual-stack-for-cf.md

```diff
@@ -0,0 +1,200 @@
+
+Furthermore, the goal is to support IPv6 for both ingress and egress traffic, ensuring that Cloud Foundry can handle inbound and outbound network communication using IPv6, further enhancing compatibility and future-proofing the platform.
+
+## Problem
+
+IPv6 support in Cloud Foundry has only reached a few CF components so far. Furthermore, the existing support often only supports one type of networking exclusively.
+
+Additionally, there are areas in the CF design informed by the limited address space of IPv4:
+
+* Workloads in Diego are interconnected via an overlay network managed by the Silk CNI. An overlay network is primarily needed to provide enough address space for the apps running within Diego and offer a network that spans all workloads.
+* On hyperscalers, egress traffic from apps requires a NAT gateway. Providing public IP addresses for all Diego cells is often not economical.
+* BOSH allows assigning a single IP address per BOSH network entry for a VM, not CIDR ranges.
+
+Adding IPv6 support should not transfer those same issues onto the IPv6 implementation.
+
+## Proposal
+
+The goal is to use IPv6 best practices and benefits to the full extent with the following high-level goals:
+
+1. NAT and port forwarding are avoided wherever possible when using IPv6
+2. Application workloads have their own IPv6 address. This allows better correlation between requests and their origin. It is important to note that application workloads shall not be addressable directly from external networks. The ingress and load balancing via Gorouter must remain.
+3. Support of dual stack networking, i.e. IPv4 and IPv6 at the same time. This is primarily for a smoother migration because the IPv6 changes can then be additive.
+4. IPv6 is additive to IPv4 functionality
+
+The ultimate future goal would be to move to pure IPv6 communication but is not a current priority and is not covered in this proposal.
+
+Backward compatibility is an important topic. Any changes that would change the default behavior should be placed behind appropriate configuration flags and turned off by default. The absence of IPv6 specific configuration must also remain without side effects to existing CF foundations.
+
+### Overview
+
+The largest conceptual change is providing individual addressing for application instances. Diego is a container runtime that is closely integrated with various other components. By extending the addressing schema to per-container addresses, all components that touch application workload information need to be adjusted.
+
+The baseline dual stack support for networks deployed via BOSH is another pillar of the proposed change.
+
+The following sections outline the changes necessary for various core components of  a typical CF foundation.
+
+### BOSH
```

Comment:
Doesn't BOSH support ipv6 already if CPI supports it? - https://bosh.io/docs/guide-ipv6-on-vsphere/

Reply:
IPv6 yes, dual stack not yet.